doc/RELEASE.adoc

```diff
@@ -0,0 +1,120 @@
+= Plutus Release Process
+:toc: left
+:reproducible:
+:figure-caption!:
+
+== Packages
+
+The following packages are versioned and released:
+
+- Libraries:
+* plutus-core:plutus-core
+* plutus-ledger-api:plutus-ledger-api
+* plutus-tx:plutus-tx
+* plutus-tx-plugin:plutus-tx-plugin
+- Executables:
+* plutus-core:pir
+* plutus-core:plc
+* plutus-core:uplc
+
+.Dependencies among Plutus packages
+image::plutus-packages.png[Plutus packages,500]
+
+== Version Scheme
+
+All above packages are versioned and released in sync. That is, whenever one package releases version X, so do all other packages. This greatly simplifies the release process compared to versioning and releasing the pacakges independently.
+
+We follow https://semver.org[semantic versioning] for the packages, where version numbers have a `major.minor.patch` pattern with the following semantics:
+
+- A major release (e.g., 3.5.2 -> 4.0.0) may contain arbitrary changes to the API (subject to the backwards compatibility requirements described below).
+- A minor release (e.g., 3.5.2 -> 3.6.0) may contain changes such as new features, which technically can be breaking, but are unlikely to cause breakage in practice, and if it does, the fix should be straightforward (e.g., by renaming).
++
+A minor release is also allowed to change the observable behaviors of functions that users should not rely on. For example, if a function returns a list of things and makes no promise on the order of elements in the returned list, then a minor release may change the order.
+- A patch release (e.g., 3.5.2 -> 3.5.3) can only contain such things as simple bug fixes, performance improvements, and documentation updates.
+
+=== Version Qualifiers
+
+Version qualifiers, also known as pre-release versions, usually refer to alpha, beta, and release candidate. Typically, alpha and beta are pre-feature complete, while release candidates are post-feature complete.
+
+We do not tag or publish alpha or beta verions for libraries. Libraries are published on Hackage, and Hackage does not support version qualifiers. It is also uncommon for a library to publish pre-release versions (unless it is new or a complete rewrite).
```

Comment:
> Libraries are published on Hackage, and Hackage does not support version qualifiers

Maybe not an argument, since you say below that we _do_ release candidates